R/utils.R

```diff
@@ -12,6 +12,7 @@ if (base::getRversion() < "3.5.0") {
 }
 isTRUEorNA    = function(x) is.logical(x) && length(x)==1L && (is.na(x) || x)
 isTRUEorFALSE = function(x) is.logical(x) && length(x)==1L && !is.na(x)
+allNA = function(x) .Call(C_allNAR, x)
```

Comment:
just ran a quick benchmark and we blow `all(is.na())` out of the water (as expected when `FALSE` but we're also faster when `TRUE` and for moderate vector size).

Worth patching to `base`? It seems not as popular to use as `anyNA` in `base` internal code:

```
grep -Er "all\s*\(is\.na\s*\(" ~/github/r-source --exclude=*tests* | wc -l
# 28, 63 if we include tests code
grep -Er "anyNA\s*\(" ~/github/r-source --exclude=*tests* | wc -l
# 190, 197 if we include tests code
```

Reply:
(as a TODO after release of course)